hoa-frontend/src/app/Services/authentication.service.ts

```diff
@@ -0,0 +1,19 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthenticationService {
+  private loggedIn = false;
+
+  constructor() {
+    const token = localStorage.getItem('token');
+    if (token) {
+      this.loggedIn = true;
+    }
+  }
+
+  isLoggedIn(): boolean {
+    return this.loggedIn;
+  }
+}
```

Comment:
@princh-canal tried to use this service to render my home-page component in the app component as what you advised but it didnt work. Did I miss something here?

Reply:
it's probably because you don't set the `isLoggedIn` to true after logging in. i'd recommend that you use a `BehaviorSubject` to get an update the `isLoggedIn` value.

then after calling your login function, also set the `isLoggedIn` value using the `AuthenticationService`